epinorm/norm.py

```diff
@@ -320,27 +474,170 @@ def _add_missing_columns(self):
         self._data["longitude"] = None
         self._data["original_record_id"] = None
 
+    def _search_tokens_diff_order(self, areas, second):
+        """
+        This method uses the geocoder to make a query of areas (a list of tokens) and second (a string).
+        we vary the order of the tokens in areas as they could be wrong. We also remove them if we still don't have a result
+        it tried all permutations up to the point where it doesn't include tokens from areas anymore (only "second")
+        """
+
+        for l in range(len(areas), -1, -1):
+            for option in itertools.permutations(areas, l):
```

Comment:
Ambiguous variable name (E741). Use "i" for iteration counter, instead of "l".

```suggestion
        for i in range(len(areas), -1, -1):
            for option in itertools.permutations(areas, i):
```